From 793a86222515770edbdd9070a6cee48d0e89c443 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 5 Nov 2014 13:40:31 -0800
Subject: [PATCH 0489/1566] drm: xilinx: dp: Propagate the error code from
 devm_clk_get()

devm_clk_get() returns the error code other than -EPROBE_DEFER.
Return the error code from devm_clk_get() instead of always returning
-EPROBE_DFERE when devm_clk_get() fails.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index b466800..52f5441 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -1251,7 +1251,7 @@ static int xilinx_drm_dp_probe(struct platform_device *pdev)
 
 	dp->aclk = devm_clk_get(dp->dev, NULL);
 	if (IS_ERR(dp->aclk))
-		return -EPROBE_DEFER;
+		return PTR_ERR(dp->aclk);
 
 	ret = clk_prepare_enable(dp->aclk);
 	if (ret) {
-- 
1.7.5.4

